bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java

```diff
@@ -0,0 +1,133 @@
+
+    public RegenerateInterleavedStorageIndexFileCommand() {
+        this(new RISIFFlags());
+    }
+
+    private RegenerateInterleavedStorageIndexFileCommand(RISIFFlags flags) {
+        super(CliSpec.<RISIFFlags>newBuilder()
+                  .withName(NAME)
+                  .withDescription(DESC)
+                  .withFlags(flags)
+                  .build());
+    }
+
+    /**
+     * Flags for regenerate interleaved storage index file command.
+     */
+    @Accessors(fluent = true)
+    @Setter
+    public static class RISIFFlags extends CliFlags {
+
+        @Parameter(names = { "-p", "--password" },
+            description = "The bookie stores the password in the index file, so we need it to regenerate."
+                          + "This must match the value in the ledger metadata.")
+        private String password = DEFAULT;
+
+        @Parameter(names = { "-b", "--b64password" },
+            description = "The password in base64 encoding, for cases where the password is not UTF-8.")
+        private String b64Password = DEFAULT;
+
+        @Parameter(names = { "-d", "--dryrun" }, description = "Process the entryLogger, but don't write anthing.")
+        private boolean dryRun;
+
+        @Parameter(names = { "-l", "--ledgerids" },
+            description = "Ledger(s) whose index needs to be regenerated. Multiple can be specified, comma separated.",
+            splitter = CommaParameterSplitter.class)
+        private List<Long> ledgerIds;
+
+    }
+
+    @Override
+    public boolean apply(ServerConfiguration conf, RISIFFlags cmdFlags) {
+        try {
+            return generate(conf, cmdFlags);
+        } catch (Exception e) {
+            throw new UncheckedExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private boolean generate(ServerConfiguration conf, RISIFFlags flags) throws NoSuchAlgorithmException, IOException {
+        checkNPE(flags);
+        byte[] password;
+        if (!flags.password.equals(DEFAULT)) {
+            password = flags.password.getBytes(StandardCharsets.UTF_8);
+        } else if (!flags.b64Password.equals(DEFAULT)) {
+            password = Base64.getDecoder().decode(flags.b64Password);
+        } else {
+            LOG.error("The password must be specified to regenerate the index file");
+            return false;
+        }
+
+        Set<Long> ledgerIds = flags.ledgerIds.stream().collect(Collectors.toSet());
+
+        LOG.info("=== Rebuilding index file for {} ===", ledgerIds);
+        ServerConfiguration serverConfiguration = new ServerConfiguration(conf);
+        InterleavedStorageRegenerateIndexOp i = new InterleavedStorageRegenerateIndexOp(serverConfiguration, ledgerIds,
+                                                                                        password);
+        i.initiate(flags.dryRun);
+
+        LOG.info("-- Done rebuilding index file for {} --", ledgerIds);
+        return true;
+    }
+
+    private void checkNPE(RISIFFlags flags) {
```

Comment:
nit : a better name will be `validateFlags`

Reply:
ok, thanks